src/range.rs

```diff
@@ -37,48 +37,82 @@ where
     }
 }
 
-pub struct Recognize<P>(P);
+pub type Recognize<P> = recognize::Recognize<P>;
+parser!{
+    #[derive(Clone)]
+    pub struct Recognize;
+    /// Zero-copy parser which reads a range of length `i.len()` and succeds if `i` is equal to that
+    /// range.
+    ///
+    /// ```
+    /// # extern crate combine;
+    /// # use combine::range::recognize;
+    /// # use combine::char::letter;
+    /// # use combine::*;
+    /// # fn main() {
+    /// let mut parser = recognize(skip_many1(letter()));
+    /// assert_eq!(parser.parse("hello world"), Ok(("hello", " world")));
+    /// assert!(parser.parse("!").is_err());
+    /// # }
+    /// ```
+    pub fn recognize[P](parser: P)(P::Input) -> <P::Input as StreamOnce>::Range
+    where [
+        P: Parser,
+        P::Input: RangeStream,
+        <P::Input as StreamOnce>::Range: ::primitives::Range,
+    ]
+    {
+        ::range::recognize_with_value(parser).map(|(range, _)| range)
+    }
+}
+
+pub struct RecognizeWithValue<P>(P);
```

Comment:
None of the other parsers derive Debug. I never bothered to add it since 95% of all parsers should be zero-sized objects (those that have size just have a function pointer or possibly a token (char, u8 ...)